sql-api/src/main/java/io/cdap/pipeline/sql/api/core/enums/OperandType.java

```diff
@@ -16,11 +16,23 @@
 
 package io.cdap.pipeline.sql.api.core.enums;
 
+import io.cdap.pipeline.sql.api.core.Column;
 import io.cdap.pipeline.sql.api.core.Filter;
+import io.cdap.pipeline.sql.api.core.interfaces.Constant;
 
 /**
  * An enum representing the operand types in a {@link Filter} expression.
  */
 public enum OperandType {
-  CONSTANT, COLUMN
+  CONSTANT(Constant.class), COLUMN(Column.class);
+
+  private final Class operandClass;
+
+  OperandType(Class operandClass) {
+    this.operandClass = operandClass;
+  }
+
+  public Class getOperandClass() {
```

Comment:
Since the enums directly link the operand types to a particular class, it seemed cleaner to just get the class from the enum instead of creating switch statements everywhere. Is there a better pattern for handling this scenario?